api4/post_test.go

```diff
@@ -1956,60 +1956,71 @@ func TestGetPost(t *testing.T) {
 	th.TestForAllClients(t, func(t *testing.T, c *model.Client4) {
 		t.Helper()
 
-		post, resp, err := c.GetPost(th.BasicPost.Id, "")
+		post, resp, err := c.GetPost(th.BasicPost.Id, "", false)
 		require.NoError(t, err)
 
 		require.Equal(t, th.BasicPost.Id, post.Id, "post ids don't match")
 
-		post, resp, err = c.GetPost(th.BasicPost.Id, resp.Etag)
+		post, resp, err = c.GetPost(th.BasicPost.Id, resp.Etag, false)
 		require.NoError(t, err)
 		CheckEtag(t, post, resp)
 
-		_, resp, err = c.GetPost("", "")
+		_, resp, err = c.GetPost("", "", false)
 		require.Error(t, err)
 		CheckNotFoundStatus(t, resp)
 
-		_, resp, err = c.GetPost("junk", "")
+		_, resp, err = c.GetPost("junk", "", false)
 		require.Error(t, err)
 		CheckBadRequestStatus(t, resp)
 
-		_, resp, err = c.GetPost(model.NewId(), "")
+		_, resp, err = c.GetPost(model.NewId(), "", false)
 		require.Error(t, err)
 		CheckNotFoundStatus(t, resp)
 
 		client.RemoveUserFromChannel(th.BasicChannel.Id, th.BasicUser.Id)
 
 		// Channel is public, should be able to read post
-		_, _, err = c.GetPost(th.BasicPost.Id, "")
+		_, _, err = c.GetPost(th.BasicPost.Id, "", false)
 		require.NoError(t, err)
 
 		privatePost = th.CreatePostWithClient(client, th.BasicPrivateChannel)
 
-		_, _, err = c.GetPost(privatePost.Id, "")
+		_, _, err = c.GetPost(privatePost.Id, "", false)
 		require.NoError(t, err)
 	})
 
 	client.RemoveUserFromChannel(th.BasicPrivateChannel.Id, th.BasicUser.Id)
 
 	// Channel is private, should not be able to read post
-	_, resp, err := client.GetPost(privatePost.Id, "")
+	_, resp, err := client.GetPost(privatePost.Id, "", false)
 	require.Error(t, err)
 	CheckForbiddenStatus(t, resp)
 
 	// But local client should.
-	_, _, err = th.LocalClient.GetPost(privatePost.Id, "")
+	_, _, err = th.LocalClient.GetPost(privatePost.Id, "", false)
 	require.NoError(t, err)
 
 	client.Logout()
 
 	// Normal client should get unauthorized, but local client should get 404.
-	_, resp, err = client.GetPost(model.NewId(), "")
+	_, resp, err = client.GetPost(model.NewId(), "", false)
 	require.Error(t, err)
 	CheckUnauthorizedStatus(t, resp)
 
-	_, resp, err = th.LocalClient.GetPost(model.NewId(), "")
+	_, resp, err = th.LocalClient.GetPost(model.NewId(), "", false)
 	require.Error(t, err)
 	CheckNotFoundStatus(t, resp)
+
+	// Normal client should get unauthorized when trying to access deleted post.
+	_, resp, err = client.GetPost(th.BasicPost.Id, "", true)
+	require.Error(t, err)
+	CheckUnauthorizedStatus(t, resp)
+
+	// System client should be able to access deleted post.
+	th.SystemAdminClient.DeletePost(th.BasicPost.Id)
+	post, _, err := th.SystemAdminClient.GetPost(th.BasicPost.Id, "", true)
+	require.NoError(t, err)
+	require.Equal(t, th.BasicPost.Id, post.Id)
```

Comment:
Could we add a test here for a normal user getting a deleted post with includeDeleted set to false? According to the Jira ticket, this should return a 404